third_party/openthread/CMakeLists.txt

```diff
@@ -26,47 +26,50 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
+set(OT_15_4 ON CACHE STRING "enable 802.15.4 radio link" FORCE)
+set(OT_ANYCAST_LOCATOR ON CACHE STRING "enable anycast locator" FORCE)
+set(OT_BACKBONE_ROUTER ${OTBR_BACKBONE_ROUTER} CACHE STRING "Enable Backbone Router feature in OpenThread" FORCE)
 set(OT_BACKBONE_ROUTER_DUA_NDPROXYING ${OTBR_DUA_ROUTING} CACHE STRING "Configure DUA ND Proxy feature in OpenThread" FORCE)
 set(OT_BORDER_AGENT ON CACHE STRING "enable border agent" FORCE)
 set(OT_BORDER_ROUTER ON CACHE STRING "enable border router feature" FORCE)
 set(OT_BORDER_ROUTING ${OTBR_BORDER_ROUTING} CACHE STRING "enable border routing feature" FORCE)
 set(OT_BORDER_ROUTING_COUNTERS ${OTBR_BORDER_ROUTING_COUNTERS} CACHE STRING "enable border routing counters feature" FORCE)
 set(OT_BUILD_EXECUTABLES OFF CACHE STRING "disable building executables" FORCE)
 set(OT_BUILTIN_MBEDTLS_MANAGEMENT OFF CACHE STRING "diable mbedTLS management" FORCE)
+set(OT_CHILD_SUPERVISION ON CACHE STRING "enable child supervision" FORCE)
 set(OT_COAP ON CACHE STRING "Enable CoAP in OpenThread")
 set(OT_COAPS ON CACHE STRING "Enable secure CoAP in OpenThread")
 set(OT_COMMISSIONER ON CACHE STRING "enable commissioner")
 set(OT_DAEMON ON CACHE STRING "enable daemon mode" FORCE)
+set(OT_DATASET_UPDATER ON CACHE STRING "enable dataset updater" FORCE)
+set(OT_DNS_CLIENT ON CACHE STRING "enable DNS client" FORCE)
```

Comment:
I'm not sure if DNS client is a requirement for a border router. Will border router itself directly use this feature?